storage_service/locations/tests/test_rclone.py

```diff
@@ -0,0 +1,464 @@
+
+
+@pytest.fixture
+def rclone_space(db):
+    space = models.Space.objects.create(
+        uuid=RCLONE_SPACE_UUID,
+        access_protocol="RCLONE",
+        staging_path="rclonestaging",
+    )
+    rclone_space = models.RClone.objects.create(
+        space=space, remote_name="testremote", container="testcontainer"
+    )
+    return rclone_space
+
+
+@pytest.fixture
+def rclone_space_no_container(db):
+    space = models.Space.objects.create(
+        uuid=RCLONE_SPACE_UUID,
+        access_protocol="RCLONE",
+        staging_path="rclonestaging",
+    )
+    rclone_space = models.RClone.objects.create(
+        space=space, remote_name="testremote", container=""
+    )
+    return rclone_space
+
+
+@pytest.fixture
+def rclone_aip(db):
+    space = models.Space.objects.create(
+        uuid=RCLONE_SPACE_UUID,
+        access_protocol="RCLONE",
+        staging_path="rclonestaging",
+    )
+    models.RClone.objects.create(
+        space=space, remote_name="testremote", container="testcontainer"
+    )
+    pipeline = models.Pipeline.objects.create(uuid=PIPELINE_UUID)
+    aipstore = models.Location.objects.create(
+        uuid=RCLONE_AS_LOCATION_UUID, space=space, purpose="AS", relative_path="test"
+    )
+    models.LocationPipeline.objects.get_or_create(pipeline=pipeline, location=aipstore)
+    aip = models.Package.objects.create(
+        uuid=RCLONE_AIP_UUID,
+        origin_pipeline=pipeline,
+        current_location=aipstore,
+        current_path="fixtures/small_compressed_bag.zip",
+        size=1024,
+    )
+    return aip
+
+
+@pytest.fixture
+def rclone_aip_no_container(db):
+    space = models.Space.objects.create(
+        uuid=RCLONE_SPACE_UUID,
+        access_protocol="RCLONE",
+        staging_path="rclonestaging",
+    )
+    models.RClone.objects.create(space=space, remote_name="testremote", container="")
+    pipeline = models.Pipeline.objects.create(uuid=PIPELINE_UUID)
+    aipstore = models.Location.objects.create(
+        uuid=RCLONE_AS_LOCATION_UUID, space=space, purpose="AS", relative_path="test"
+    )
+    models.LocationPipeline.objects.get_or_create(pipeline=pipeline, location=aipstore)
+    aip = models.Package.objects.create(
+        uuid=RCLONE_AIP_UUID,
+        origin_pipeline=pipeline,
+        current_location=aipstore,
+        current_path="fixtures/small_compressed_bag.zip",
+        size=1024,
+    )
+    return aip
+
+
+def test_rclone_delete(mocker, rclone_aip):
+    """Mock method call and assert correctness of rclone command."""
+    delete_path = mocker.patch("locations.models.rclone.RClone._execute_subprocess")
+    mocker.patch(
+        "locations.models.rclone.RClone.remote_prefix",
+        return_value="testremote:",
+        new_callable=mocker.PropertyMock,
```

Comment:
TIL `PropertyMock` :open_mouth:

Reply:
This one took me a while to figure out! :laughing: